src/Common/src/System/Net/SocketAddress.cs

```diff
@@ -228,16 +228,18 @@ public override int GetHashCode()
 
         public override string ToString()
         {
-            StringBuilder bytes = new StringBuilder();
-            for (int i = SocketAddressPal.DataOffset; i < this.Size; i++)
+            var sb = new StringBuilder().Append(Family.ToString()).Append(':').Append(Size).Append(":{");
+
+            for (int i = SocketAddressPal.DataOffset; i < Size; i++)
             {
                 if (i > SocketAddressPal.DataOffset)
                 {
-                    bytes.Append(",");
+                    sb.Append(',');
                 }
-                bytes.Append(this[i].ToString(NumberFormatInfo.InvariantInfo));
+                sb.Append(this[i]);
             }
-            return Family.ToString() + ":" + Size.ToString(NumberFormatInfo.InvariantInfo) + ":{" + bytes.ToString() + "}";
+
+            return sb.Append('}').ToString();
```

Comment:
It seems in the method we could use string.Create() and exclude allocations at all.

Reply:
We could, though unless you got really complicated it would likely still require the Family.ToString() in order to know exactly how big a string to allocate.  Feel free to submit a PR.

Reply:
It seems `Family.ToString()` doesn't allocate. Key problem is ` sb.Append(this[i]);`. It can return from 1 to 3 chars. And we have to pass twice that complicates the code too much.
So I see only one thing we could do - explicit set init size:
```c#
            var sb = new StringBuilder(148).Append(Family.ToString()).Append(':').Append(Size).Append(":{");
```